File: bubble_sort/python/bubble.py

def bubble_sort(arr:list):
    n = len(arr) 

    # Traverse through all array elements 
    for i in range(n): 
  
        # Last i elements are already in place 
        for j in range(0, n-i-1): 
  
            # traverse the array from 0 to n-i-1 
            # Swap if the element found is greater 
            # than the next element 
            if arr[j] > arr[j+1] : 
                arr[j], arr[j+1] = arr[j+1], arr[j] 
  
    
    print('ordered: ',arr)
        


if __name__ == '__main__':
    # declare variables
    print("introduce numbers to sort")
    n = int(input())
    numbers = list()

    #read numbers
    for i in range(n):
        print(f"introduce n{i+1}: ")
        numbers.append(input())

    #print unordered numbers
    print('unordered: ',numbers)

    # order
    bubble_sort(numbers)
        #iterate numbers and
            #validate max value.
                #if 
                    #swap
                #else 
                    #ok
    # print ordered

File: bubble_sort/go/bubble.go
package main

import "fmt"

func bubbleSort(list []uint) {
	for i := range list {
		for j := 0; j < len(list)-i-1; j++ {
			if list[j] > list[j+1] {
				swap := list[j]
				list[j] = list[j+1]
				list[j+1] = swap

			}
		}
	}
}

func main() {
	list := []uint{41, 22, 35, 11, 5, 99982, 3214, 52217, 1, 321, 43, 77, 32, 11, 25, 74, 78, 789, 123, 124, 357, 951, 741, 369, 32, 323, 321, 324, 3217, 3213697, 1152, 1, 12, 574, 37, 21}
	bubbleSort(list)
	fmt.Println(list)
}
